### ISSUE TITLE
SimpleJPARepository: Methods accessing metadata.getComment() without Null-Check produce NPE

### ISSUE THREAD

**Issue by jpeschke2608** [spring-projects/spring-data-jpa#2716]
Hi there,
Stumbled upon a Null Pointer Exception during Migration to Spring Boot 3.0.0 (and spring-data-jpa 3.0.0) that seems to be raised when accessing the "getComment" method on the metadata field of SimpleJPARepository:

The two methods "applyQueryHints" and "applyQueryHintsForCount()" don't perform a null check on the metadata field of SimpleJPARepository (which is annotated as @Nullable), e.g.:

```
private void applyQueryHintsForCount(Query query) {
	getQueryHintsForCount().forEach(query::setHint);

       // NPE is thrown here, if metadata is null!
	if (metadata.getComment() != null && provider.getCommentHintKey() != null) {
		query.setHint(provider.getCommentHintKey(), provider.getCommentHintValue(metadata.getComment()));
	}
}
```
In other methods of the repository, it seems to be correct, e.g.:
```
private <S> TypedQuery<S> applyRepositoryMethodMetadata(TypedQuery<S> query) {
	if (metadata == null) {
	   return query;
	}
	LockModeType type = metadata.getLockModeType();
       ...
}	
```
Thanks for any ideas for smooth workarounds or other input,

Joerg

**Comment by gregturn** [comment]
I added additional `null` checks in `applyQueryHints` and `applyQueryHintsForCount` to further guard against this.

**Comment by dickerpulli** [comment]
@gregturn Unfortunately the commits https://github.com/spring-projects/spring-data-jpa/commit/f40b39c3ec1039c36f4545ee9a5f74c6451d850a and https://github.com/spring-projects/spring-data-jpa/commit/b0aa2b67546698fefc4218393ea9eebc01524c69 has not been pushed to the 3.0.x branch, so they were not a part of the 3.0.1 release. Could you please add them to the 3.0.x branch, so that they are at least included in next release 3.0.2?

Thanks.

**Comment by gregturn** [comment]
I believe this was merged to `main` right after `main` rolled to `3.1.x`. I've backported this fix to `3.0.x` via 7dce3d119fa1e8a01d0e29fca1b790dbf6b19383.
